### ISSUE TITLE
Deadlock in delegate which created used 

### ISSUE THREAD

**Issue by Rombersoft** [mono/mono#6573]
## Steps to Reproduce

1. Assembly computer with motherboard msi 760GM-p23. On other motherboards all works fine!
2. Install 4.8.8-2-ARCH #1 SMP PREEMPT Thu Nov 17 15:09:32 CET 2016 i686 GNU/Linux
3. Install mono 5.8
4. git clone from https://github.com/MonoGame/MonoGame
5. Create project monogame with some animation

## Expected Behavior
Application must draw animation

## Description and other observation
Monogame is game engine which in cycle runs methods Draw and Update. 
At start I thought that problem in monogame, I even create topic here http://community.monogame.net/t/drawing-stops/9799
But when I tried with old version monogame (2.5) on mono 5.4 version that problem still exists.
But when I tried with old version monogame (2.5) on mono 2.10 version that problem does not exist.
I can not use monogame 3.6 on mono 2.10 because 2.10 version does not support 4.5 framework and monogame 3.6 does not work on 4.0 framework

### On which platforms did you notice this

[ ] macOS
[*] Linux
[ ] Windows

**Version Used**:

### Stacktrace

I can not reproduce Stack trace. But I know place where drawing thread stops.
Look at here
https://github.com/MonoGame/MonoGame/blob/master/MonoGame.Framework/Graphics/GraphicsDevice.OpenGL.cs#L356
I am sure that it is problem on new mono.

**Comment by Rombersoft** [comment]
Please read closely. I've written that old monogame works on old mono version 2.10. Issue is not that animation stops, issue is that thread stops in call delegate while action for inputs (keyboard, mouse) is missing.

**Comment by marek-safar** [comment]
That still does not give me any clue what "thread stops in call delegate" mean

**Comment by Rombersoft** [comment]
Marek are you alone mono developer? Have you tried to reproduce my case in order to understand me? Mono was super mono when Novel existed but now when Xamarin created Xamarin.Forms and Xamarin.Android mono isn't interesting.

**Comment by Rombersoft** [comment]
I can create simple project for your checking.

**Comment by luhenry** [comment]
@Rombersoft yes that would be most helpful, and please attach with the project the different steps we need to execute to reproduce. Thank you very much!

**Comment by Rombersoft** [comment]
This link to download simple project monogame (450 Mb)

https://drive.google.com/open?id=1n6tqr3fZsDpdvSePtzWWDVsu6J3Y9z1U
